### PR TITLE
Improve grid rotation

### PR DESCRIPTION
**Pull request by rubencalje**

### CODE REVIEW

File: nlmod/dims/base.py
Comment (OnnoEbbens) [inline nlmod/dims/base.py:1]: the docstrings of `get_ds` and `_set_angrot_attributes` mention for "xorigin" and "yorigin": "Only used when angrot is not 0." I think this is incorrect and they are also used if angrot is 0. It would be nice to have a bit more information in the docstring about what happens to the extent when you use `xorigin`, `yorigin` and `angrot`. What coordinates are there in the dataset model coördinates, real-world-coordinates.

File: nlmod/dims/grid.py [hunk 1]
```diff
@@ -1536,7 +1542,7 @@ def gdf_to_count_da(gdf, ds, ix=None, buffer=0.0, **kwargs):
 
     # build list of gridcells
     if ix is None:
-        modelgrid = modelgrid_from_ds(ds)
+        modelgrid = modelgrid_from_ds(ds, rotated=False)
```
Comment (OnnoEbbens) [inline nlmod/dims/grid.py:1545]: Just checking if I understand correctly: You convert the gdf to modelcoordinates if you have a rotated grid. Then you use `rotated=False` for the modelgrid because you need a modelgrid with modelcoordinates (and not real world coördinates).

File: nlmod/dims/resample.py [hunk 2]
```diff
@@ -139,11 +139,11 @@ def ds_to_structured_grid(
     if delc is None:
         delc = delr
```
Comment (OnnoEbbens) [inline nlmod/dims/resample.py:141]: see previous comment about xorigin and yorigin in docstring.

File: nlmod/dims/resample.py [hunk 3]
```diff
@@ -431,7 +436,9 @@ def dim_to_regular_dim(da, dims, z):
         coords = dict(da.coords)
         coords["x"] = ds.x
         coords["y"] = ds.y
-        coords.pop("icell2d")
+        for key in list(coords):
```
Comment (OnnoEbbens) [inline nlmod/dims/resample.py:439]: Why is this necessary? I think 'icell2d' is always in the coordinates of da because at the beginning of the function this checked.

File: tests/test_026_grid.py
Comment (OnnoEbbens) [inline tests/test_026_grid.py:1]: Looks neat!

File: nlmod/dims/base.py
Comment (rubencalje) [reply nlmod/dims/base.py:1]: There is a notebook for that: https://nlmod.readthedocs.io/en/latest/examples/11_grid_rotation.html

File: nlmod/dims/grid.py (hunk 1 above)
Comment (rubencalje) [reply nlmod/dims/grid.py:1545]: Yes

File: nlmod/dims/resample.py (hunk 3 above)
Comment (rubencalje) [reply nlmod/dims/resample.py:439]: We need to delete all coordinates that have icell2d as a dimension, like 'xc', 'yc'. The coordinate 'icell2d' also has 'icell2d' as a dimension, so it is also removed. So no need to specifically delete it, this saves a line.

File: nlmod/dims/base.py
Comment (rubencalje) [reply nlmod/dims/base.py:1]: You are right, xorigin and yorigin are used when angrot == 0. Will change the docstrings.

**Comment by rubencalje** [comment]
I implemented the requested changes, by improving the docstring of xorigin, yorigin and angrot.

File: nlmod/dims/resample.py (hunk 3 above)
Comment (OnnoEbbens) [reply nlmod/dims/resample.py:452]: Ah that makes sense, thanks for explaining

**Review by OnnoEbbens — APPROVED** [review]
